experimental/rules_python/wheelmaker.py

```diff
@@ -157,13 +158,38 @@ def add_metadata(self, extra_headers, description, classifiers, requires,
         metadata += "\n"
         self.add_string(self.distinfo_path('METADATA'), metadata)
 
-    def add_entry_points(self, console_scripts):
+    def add_entry_points(self, entry_points):
         """Write entry_points.txt file to the distribution."""
         # https://packaging.python.org/specifications/entry-points/
-        if not console_scripts:
+        import re
+        pattern = re.compile(r'([^;]+);([^=]+)=(.*)') # group;name=ref
+        if not entry_points:
             return
-        lines = ["[console_scripts]"] + console_scripts
-        contents = '\n'.join(lines)
+
+        # Reassemble the entry_points array into
+        groups = {}
+        for line in entry_points:
+            match = pattern.fullmatch(line)
+            if not match:
+                raise ValueError('{line} is not a valid entry point'.format(line=line))
+            group = match.group(1).strip()
+            group_dict = groups.setdefault(group, {})
+            name = match.group(2).strip()
+            object_reference = match.group(3).strip()
+            if name in group_dict:
+                raise ValueError("Duplicate entry for name {name} in group {group}".format(name=name, group=group))
```

Comment:
Is there a stock formatter for this repo? I tried `yapf` but it changed way more than just the modified code.